chips/cc2650/src/ioc.rs

```diff
@@ -0,0 +1,115 @@
+/* IOC - IO Configuration */
+
+use kernel::common::VolatileCell;
+use kernel::hil;
+
+pub const IOC_PULL_CTL: u8 = 13;
+pub const IOC_IE: u8 = 29;
+pub const IOC_EDGE_DET: u8 = 16;
+pub const IOC_EDGE_IRQ_EN: u8 = 18;
+
+#[repr(C)]
+struct IOC {
+    iocfg: [VolatileCell<u32>; 32],
+}
+
+pub const IOC_BASE: usize = 0x4008_1000;
+
+#[allow(non_snake_case)]
+fn IOC() -> &'static IOC {
+    unsafe { &*(IOC_BASE as *const IOC) }
+}
+
```

Comment:
I have the same question about the struct `structure` and the ```standalone non_snake_case function```. See GpioPin

Reply:
We followed the lead of `nrf5x` in the use of a standalone function. Maybe it'd be good to change all occurences in the code base at once - if a change is desirable?

Reply:
Yeah, I know but I think we should strive to use methods over functions and try to use the Rust coding guidelines accordingly. But what does the @helena-project/core-team think?

Reply:
Yeah, in general I agree that it would be better to have registers included as a struct member and only access it in internal methods

(ex: https://github.com/helena-project/tock/blob/master/chips/sam4l/src/adc.rs#L105
        https://github.com/helena-project/tock/blob/master/chips/sam4l/src/adc.rs#L170
        https://github.com/helena-project/tock/blob/master/chips/sam4l/src/adc.rs#L213)

Is there a reason why the IOC registers would need to be accessed outside of the IOC struct's methods? Other drivers may not do the right thing, but that's no reason not to here.

(Also, you are part of core-team, Niklas :P)

Reply:
Not really :) imho the syntax of doing like [adc.rs you linked](https://github.com/helena-project/tock/blob/master/chips/sam4l/src/adc.rs#L105) is way nicer!

I'll update the PR with changes to remove `GPIO()` and `IOC()` :)